web/src/title/LoadSavedProject.svelte

```diff
@@ -12,24 +13,52 @@
 
     let contents = await fileInput.files![0].text();
     let gj = JSON.parse(contents);
-    // Is this a CNT project or a regular one?
-    let key = "";
+
+    let projectName: string;
+
+    let studyAreaName: string = "";
+
+    // Legacy loading logic, this can go away at some point if we want to drop support for old saved file formats.
+
+    // Is this a CNT project or a global one?
     if (gj.study_area_name && gj.study_area_name.startsWith("LAD_")) {
       let kind = "ltn_cnt";
       // Parse the project name from the filename, best effort. The user may
       // have renamed the file.
-      let projectName = stripSuffix(
+      projectName = stripSuffix(
         stripPrefix(filename, `${kind}_${gj.study_area_name}_`),
         ".geojson",
       );
-      key = `${kind}/${gj.study_area_name}/${projectName}`;
     } else {
-      let projectName = stripSuffix(stripPrefix(filename, "ltn_"), ".geojson");
-      key = `ltn_${projectName}`;
+      projectName = stripSuffix(stripPrefix(filename, "ltn_"), ".geojson");
+    }
+
+    if (gj.study_area_name) {
+      studyAreaName = gj.study_area_name;
     }
-    // TODO Be careful with overwriting files
```

Comment:
We now increment a number when importing a file with an existing name. With the new UUID logic they wouldn't get "clobbered", but for sanity, there is some application logic which ensures names must be unique upon save.